src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPattern.java

```diff
@@ -1,102 +1,117 @@
-        this.structural = structural;
-    }
-
-    public boolean isBehavioral() {
-        return behavioral;
-    }
-
-    public void setBehavioral(boolean behavioral) {
-        this.behavioral = behavioral;
-    }
-
-    public String getPatternCode() {
-        return patternCode;
-    }
+    public String refactor(String badCode){
+        /* some magic involving patternCode happens here, irrelevant from this example perspective...*/
 
-    public void setPatternCode(String patternCode) {
-        this.patternCode = patternCode;
+        String goodCode = "this code is very good";
+        return goodCode;
     }
 
-    public DesignPattern(String id, String name, String description, String example, String patternCode, boolean creational, boolean structural, boolean behavioral) {
-        this.id = id;
-        this.name = name;
-        this.description = description;
-        this.example = example;
+    private DesignPattern(Details patternDetails, String patternCode) {
+        this.patternDetails = patternDetails;
         this.patternCode = patternCode;
-        this.creational = creational;
-        this.structural = structural;
-        this.behavioral = behavioral;
     }
 
-    public String refactor(String badCode){
-        /* some magic involving patternCode happens here, irrelevant from this example perspective...*/
-
-        String goodCode = "this code is very good";
-        return goodCode;
+    @Component
+    public static class Builder implements CreatingDesignPattern, HavingName {
+        @Override
+        public DesignPattern create() {
+            return new DesignPattern(
+                    new Details("1", name(), "builder pattern", "some builder example", Type.CREATIONAL),
+                    "some builder code"
+            );
+        }
+
+        @Override
+        public String name() {
+            return "builder";
+        }
+    }
+
+    @Component
+    public static class Decorator implements CreatingDesignPattern, HavingName {
+        @Override
+        public DesignPattern create() {
+            return new DesignPattern(
+                    new Details("2", name(), "decorator pattern", "some decorator example", Type.CREATIONAL),
+                    "some decorator code"
+            );
+        }
+
+        @Override
+        public String name() {
+            return "decorator";
+        }
+    }
+
+    @Component
+    public static class Strategy implements CreatingDesignPattern, HavingName {
+        @Override
+        public DesignPattern create() {
+            return new DesignPattern(
+                    new Details("3", name(), "strategy pattern", "some strategy example", Type.BEHAVIORAL),
+                    "some strategy code"
+            );
+        }
+
+        @Override
+        public String name() {
+            return "strategy";
+        }
+    }
+
+    private enum Type {
+        CREATIONAL, STRUCTURAL, BEHAVIORAL
+    }
+
+     static class Details {
```

Comment:
Class completely obsolete. Just make DesignPattern constructor private and allow creating objects via static classes or better static methods.